frontend/dockerfile/shell/lex_test.go

```diff
@@ -248,3 +248,40 @@ func TestProcessWithMatches(t *testing.T) {
 
 	require.Equal(t, 0, len(matches))
 }
+
+func TestProcessWithMatchesPlatform(t *testing.T) {
+	shlex := NewLex('\\')
+
+	const (
+		// corresponds to the filename convention used in https://github.com/moby/buildkit/releases
+		release = "something-${VERSION}.${TARGETOS}-${TARGETARCH}${TARGETVARIANT:+-${TARGETVARIANT}}.tar.gz"
```

Comment:
Caused me some head-scratching to understand what it was doing; https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_02

So the `:+` substitutes `TARGETVARIANT` _if it's set and not empty_, so for this case

- if `$TARGETVARIANT` is not set (or "set, but null"), nothing is returned
- if `$TARGETVARIANT` is set, it returns `-$TARGETVARIANT` (prepends a hyphen)

(Wondering if the test should be simplified to test for that specific part (or if we should have a test _just_ for the `${TARGETVARIANT:+-${TARGETVARIANT}}`), but I guess there's some value in having the whole string (as we're testing the lexer, not _only_ that form of substitution).